Bio/Align/mauve.py

```diff
@@ -0,0 +1,233 @@
+        count = interfaces.AlignmentWriter.write_file(self, alignments)
+        return count
+
+    def write_alignment(self, alignment):
+        """Use this to write (another) single alignment to an open file."""
+        n, m = alignment.shape
+
+        if n == 0:
+            raise ValueError("Must have at least one sequence")
+        if m == 0:
+            raise ValueError("Non-empty sequences are required")
+
+        filename = self._filename
+        stream = self.stream
+        for i in range(n):
+            identifier = alignment.sequences[i].id
+            start = alignment.coordinates[i, 0]
+            end = alignment.coordinates[i, -1]
+            if start <= end:
+                strand = "+"
+            else:
+                strand = "-"
+                start, end = end, start
+            if start == end:
+                assert start == 0
+            else:
+                start += 1  # switch to 1-based counting
+            sequence = alignment[i]
+            if filename is None:
+                number = (
+                    self._filenames.index(identifier) + 1
+                )  # Switch to 1-based counting
+                line = f"> {number}:{start}-{end} {strand} {identifier}\n"
+            else:
+                number = int(identifier) + 1  # Switch to 1-based counting
+                line = f"> {number}:{start}-{end} {strand} {filename}\n"
+            stream.write(line)
+            line = f"{sequence}\n"
+            stream.write(line)
+        stream.write("=\n")
+
+
+class AlignmentIterator(interfaces.AlignmentIterator):
+    """Mauve xmfa alignment iterator."""
+
+    def __init__(self, source):
+        """Create an AlignmentIterator object.
+
+        Arguments:
+         - source   - input data or file name
+
+        """
+        super().__init__(source, mode="t", fmt="Mauve")
+        stream = self.stream
+        metadata = {}
+        prefix = "Sequence"
+        suffixes = ("File", "Entry", "Format")
+        id_info = {}
+        for suffix in suffixes:
+            id_info[suffix] = []
+        for line in stream:
+            if not line.startswith("#"):
+                self._line = line.strip()
+                break
+            key, value = line[1:].split()
+            if key.startswith(prefix):
+                for suffix in suffixes:
+                    if key.endswith(suffix):
+                        break
+                else:
+                    raise ValueError("Unexpected keyword '%s'" % key)
+                if suffix == "Entry":
+                    value = int(value) - 1  # Switch to 0-based counting
+                seq_num = int(key[len(prefix) : -len(suffix)])
+                id_info[suffix].append(value)
+                assert seq_num == len(id_info[suffix])  # Mauve uses 1-based counting
+            else:
+                metadata[key] = value.strip()
+        else:
+            if not metadata:
+                raise ValueError("Empty file.") from None
```

Comment:
As in another recent ``Bio.Align`` parser PR, having an explicit empty file test seems wise.

Reply:
Added a test for an empty file.